### ISSUE TITLE
Compatibility with PopOut!

### ISSUE THREAD

**Issue by kgar** [kgar/foundry-vtt-tidy-5e-sheets#6]
Provide a level of compatibility with [PopOut!](https://foundryvtt.com/packages/popout) that makes Tidy 5e Sheets usable.
The player should be able to sheet's core functions
- tabs
- rolls
- item tables and grids
- exhaustion / inspiration trackers, etc.
- HP Bars
- currency updates - https://github.com/sdenec/tidy5e-sheet/issues/793
- dark mode / theme setting overrides - https://github.com/sdenec/tidy5e-sheet/issues/686
- item cards - when the Item Card is set to **floating**, the user should be able to see it within the popped out window. It is not possible to support the slide-out version of Item Cards when popped out

> **Note**
> Not all functionality is available when popped out. For example, when adjusting theme settings in real time, the changes will not be reflected to a popped out sheet. Also, slide-out item cards have nowhere visible to slide to, so the user would not be able to see them.

## References

https://discord.com/channels/732325252788387980/1116078321067892796/1136092315337887836

Fallayn
Hey! Reading this all it's fantastic to see this being worked on. I also want to chip in some $ if that helps in any way, just let me know where. This is the single most critical 5E module for me.

One concern I want to mention re: Svelte - one pain point I have seen with other (system) sheets that use more custom JS tech is that it often ends up incompatible with Popout for one reason or another. Would be great if the new Tidy5E would still work with the Popout module.

kgar
I will add this to the list.

To help me with knowing what to look for, what kind of breakages occur with popOut! Is it functionality, styles, and/or other?

Fallayn

Failure modes I've seen:

1. Popout does not work at all, i.e., when pressing the PopOut! button in a actor sheet's title bar, either nothing happens or the opened popout window is obviously broken (styling etc)
2. Sheet modifications either do not work, or only work in the UI, but are not persisted (so changing HP, adding an item or something either does not work, or "works" but if you close the popout window and check the actor in Foundry, the changes were not saved)
3. Rolls do not work (either not at all, dice simply aren't rolled, or the roll popup windows does not show in the first place)
4. When closing the popout or "popping it back in", data goes awry or rolls made while popped out are repeated (this is a weird one I've only seen in one system)

So basically either popping out is broken, popping in is broken, or saving data is broken. Besides that, if/once those actually work, it seems to behave like the normal sheet window, I haven't seen much PopOut specific styling issues or such.

**Comment by kgar** [comment]
Closing as tentatively completed. We are as compatible as we can be at the moment.
Tidy now also has a special plugins feature specifically for modules like PopOut! and warpgate.
